crates/sui-package-resolver/src/store.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::path::Path;
+use std::sync::Arc;
+
+use crate::error::Error;
+use crate::Result;
+use crate::{Module, Package};
+use async_trait::async_trait;
+use diesel::{ExpressionMethods, OptionalExtension, QueryDsl, RunQueryDsl};
+use move_binary_format::{access::ModuleAccess, errors::Location, CompiledModule};
+use move_core_types::account_address::AccountAddress;
+use sui_indexer::{indexer_reader::IndexerReader, schema_v2::objects};
+use sui_rest_api::Client;
+use sui_types::base_types::ObjectID;
+use sui_types::{base_types::SequenceNumber, move_package::TypeOrigin, object::Object};
+use typed_store::rocks::{DBMap, MetricConf};
+use typed_store::traits::TableSummary;
+use typed_store::traits::TypedStoreDebug;
+use typed_store::Map;
+use typed_store_derive::DBMapUtils;
+
+/// Interface to abstract over access to a store of live packages.  Used to override the default
+/// store during testing.
+#[async_trait]
+pub trait PackageStore {
+    /// Latest version of the object at `id`.
+    async fn version(&self, id: AccountAddress) -> Result<SequenceNumber>;
+
+    /// Read package contents. Fails if `id` is not an object, not a package, or is malformed in
+    /// some way.
+    async fn fetch(&self, id: AccountAddress) -> Result<Package>;
+
+    /// Store package `object` in the underlying store
+    async fn update(&self, object: &Object) -> Result<()>;
```

Comment:
this function was added to the existing trait to support underlying store with new packages